enterprise/server/remote_execution/executor/executor.go

```diff
@@ -199,15 +199,28 @@ func (s *Executor) ExecuteTaskAndStreamResults(ctx context.Context, st *repb.Sch
 		defer stop()
 	}
 
-	task := st.ExecutionTask
+	task := st.GetExecutionTask()
 	req := task.GetExecuteRequest()
 	taskID := task.GetExecutionId()
 	adInstanceDigest := digest.NewResourceName(req.GetActionDigest(), req.GetInstanceName(), rspb.CacheType_AC, req.GetDigestFunction())
 	digestFunction := adInstanceDigest.GetDigestFunction()
 	task.ExecuteRequest.DigestFunction = digestFunction
 	acClient := s.env.GetActionCacheClient()
 
-	stateChangeFn := operation.GetStateChangeFunc(stream, taskID, adInstanceDigest)
+	auxMetadata := &espb.ExecutionAuxiliaryMetadata{
+		PlatformOverrides:  task.GetPlatformOverrides(),
+		ExecutionTask:      task,
```

Comment:
do we use the ExecutionTask for anything? seems like SchedulingMetadata has everything we need?

Reply:
I'm using it to get at `execute_request.skip_cache_lookup` and `execute_request.execution_policy.priority`. Maybe I should send the ExecuteRequest instead of the ExecutionTask, since we already have access to most of its other fields:
- action and command are fetched from CAS
- execution_id is the operation name
- invocation_id and request_metadata come in the RPC context.
- platform_overrides and queued_timestamp are in the result metadata.
- jwt is the only field I'm not sure where to get.

What do you think?

Reply:
setting the ExecuteRequest sounds good to me